_posts/2018-07-13-KafkaSchemaRegistryPart1.md

```diff
@@ -0,0 +1,199 @@
+---
+layout: post
+title: Kafka - Confluent Schema Registry. Part 1
+categories: Kafka
+tags: [Kafka, Schema registry, Apache Avro, Confluent]
+author: NikitaZhevnitskiy
+---
+
+# Introduction
+
+Let's talk about Kafka Schema Registry essentials.
+
+# Repository
+
+Repository contains examples of schema evolution - full compatibility.
+[Kafka-schema-registry-poc](https://github.com/sysco-middleware/kafka-schema-registry-poc)
+
+## Content: 
+
+1. What is schema registry and why we need it?  
+1.1. Use case.  
+1.2. Full compatibility.  
+1.3. Avro serializer.
+2. How to define schema?  
+2.1. In source code.  
+2.2. Using reflection.  
+2.3. Using json.
+3. Schema evolution example.  
+3.1. Up kafka.  
+3.2. Up project.
+4. Important notes.
+
+## 1. What is schema registry and why we need it?
+
+Kafka schema registry is service(app) which stores schemas and provides RESTful interface to preform CRUD operations. 
+Schemas can be applied to key/value or both.
+Kafka schema registry is separate node. Kafka Consumer/Producer should have `schema.registry.url` in properties, 
+if schema registry is in use.
+```
+  Properties properties = new Properties();
+  properties.setProperty("schema.registry.url", "http://localhost:8081");
+```
+
+Kafka schema registry should be fault tolerant.
+Why? Short answer is to `centralize` message verification.
```

Comment:
I will disagree with this. Main reason is service depedency from clients. If is not available clients will fail. 
Maybe you mean the same by centralizing verification, but I think is worth clarifying this.

More info: <https://www.infoq.com/presentations/contracts-streaming-microservices>